packages/opentelemetry-basic-tracer/src/Span.ts

```diff
@@ -0,0 +1,158 @@
+/**
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as types from '@opentelemetry/types';
+import {
+  randomSpanId,
+  randomTraceId,
+  INVALID_SPAN_CONTEXT,
+  isValid,
+} from '@opentelemetry/core';
+import { performance } from 'perf_hooks';
+import { TraceOptions } from '@opentelemetry/types';
+
+/**
+ * This class represents a span.
+ */
+export class Span implements types.Span {
+  private readonly _spanContext: types.SpanContext = INVALID_SPAN_CONTEXT;
+  private readonly _tracer: types.Tracer;
+  private readonly _parentId?: string;
+  private readonly _kind: types.SpanKind;
+  private readonly _attributes: types.Attributes = {};
+  private readonly _links: types.Link[] = [];
+  private readonly _events: types.Event[] = [];
+  private _status: types.Status = {
+    code: types.CanonicalCode.OK,
+  };
+  private _name: string;
+  private _ended = false;
+  private _startTime: number;
+  private _endTime = 0;
```

Comment:
Didn't we agree to use duration for Node specifically because it provides higher precision as a number than an end time?

Reply:
Do you think that still applies now that we use `performance.now` for the timestamps (instead of epoch millis)?

It is true that Node processes that have been running for > ~90 days will begin to get accuracy slightly less than nanosecond accuracy for `performance.now` timestamps, but accuracy of 
10 nanos still applies for `performance.now` timestamps even for processes that have been running for 2 years (try running `2*365*24*3600*1000 + 1e-5` in a console).

My opinion though is that given this relatively high accuracy even for long running jobs that we don't worry about the workaround to treat span duration differently from its start time or the timestamps of events within a span.

See https://github.com/open-telemetry/opentelemetry-js/issues/19 for some related discussion

Reply:
Also, given that we would be calculating `duration` as the different between `performance.now` timestamps, it's not totally clear to me that we would get higher accuracy that route. We'd need to use something like `process.hrtime`, which then requires wrapping it in a platform specific code.

Reply:
Moved this conversation in a separate issue https://github.com/open-telemetry/opentelemetry-js/issues/144 to unblock this PR, will revisit again based on discussion.